### PR TITLE
TH-84: Event page hero

### PR DESCRIPTION
**Pull request by jorilindell**
https://helsinkisolutionoffice.atlassian.net/browse/TH-84

Only the hero component is implemented in scope of this PR. The task is huge so it doesn't make sense to implement everything in one PR

### CODE REVIEW

File: client/src/util/createClient.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { InMemoryCache } from "apollo-cache-inmemory";
+import ApolloClient from "apollo-client";
+import { ApolloLink } from "apollo-link";
+import { BatchHttpLink } from "apollo-link-batch-http";
+import { createPersistedQueryLink } from "apollo-link-persisted-queries";
+import get from "lodash/get";
+
+export default function createClient(uri: string) {
```
Comment (jimppa) [inline client/src/util/createClient.tsx]: There is nothing in this file indicating that this is a graphql-client, maybe rename file and function to createGraphqlClient?

Comment (jorilindell) [reply client/src/util/createClient.tsx]: Good point. Renamed

File: graphql-proxy/src/utils/toCamelCase.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export default (snakecase: string) => {
```
Comment (badrange) [inline graphql-proxy/src/utils/toCamelCase.ts]: What about adding some comments explaining what this function does? A couple of before-after examples would save a lot of brainpower.

File: graphql-proxy/src/utils/objectToCamelCase.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import memoize from "lodash/memoize";
+
+import toCamelCase from "./toCamelCase";
+
+const memoizedCamelCase = memoize(toCamelCase);
+
+const objectToCamelCase = value => {
```
Comment (badrange) [inline graphql-proxy/src/utils/objectToCamelCase.ts]: An explanation of what and why could be helpful for a future developer joining the project

File: client/src/util/getLocalisedString.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import { Language } from "../types";
+
+export default (obj: LocalizedObject, language: Language) => {
+  const languages = [
+    language,
+    ...Object.values(SUPPORT_LANGUAGES).filter(item => item !== language)
+  ];
+  let str = "";
+
+  forEach(languages, lng => {
```
Comment (jimppa) [inline client/src/util/getLocalisedString.ts]: [`Array.prototype.find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) does exactly what you are doing here

Comment (jimppa) [reply client/src/util/getLocalisedString.ts]: Also, a unit-test would be helpful documentation for this module

File: client/src/common/components/keyword/keyword.module.scss [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  background-color: $hel-summer-light-50;
+  color: $hel-black;
+  margin: 0.1875rem 0.375rem 0.1875rem 0;
+  display: inline-flex;
+  padding: 0.125rem 0.5rem;
+  align-items: center;
+  font-size: 0.875rem;
+  font-weight: 500;
+
+  @include respond_below(sm) {
```
Comment (jimppa) [inline client/src/common/components/keyword/keyword.module.scss]: Would prefer mobile first in the future

File: client/src/icons/AngleRightIcon.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+
+type Props = { className?: string };
+
+export default ({ className = "" }: Props) => (
```
Comment (jimppa) [inline client/src/icons/AngleRightIcon.tsx]: If you are using CRA, you can add `className` to svg-icons just fine. So I would have this module just as svg-module instead, no need for tsx.

File: client/src/domain/event/EventHero.tsx [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+              )}
+              <div className={classNames(styles.date, styles.desktopOnly)}>
+                {getDateStr(startTime, endTime)}
+              </div>
+              <div className={styles.title}>
+                {getLocalisedString(name, locale)}
+              </div>
+              <div
+                className={styles.description}
+                dangerouslySetInnerHTML={{
```
Comment (jimppa) [inline client/src/domain/event/EventHero.tsx]: Why does this need to be set dangerously?

File: client/src/domain/event/EventHero.tsx [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+                    keywords.map(keyword => {
+                      return (
+                        <Keyword
+                          key={keyword.id}
+                          keyword={getLocalisedString(keyword.name, locale)}
+                        />
+                      );
+                    })}
+                  {today && !thisWeek && (
+                    <Keyword color="lightEngel50" keyword="Today" />
```
Comment (jimppa) [inline client/src/domain/event/EventHero.tsx]: Should this keyword be translated?

File: client/src/domain/event/EventHero.tsx [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+    );
+    const streetAddress = getLocalisedString(
+      (location && location.streetAddress) || {},
+      locale
+    );
+
+    return [locationName, streetAddress, addressLocality].join(", ");
+  };
+
+  const getDateStr = (start: string, end: string | null | undefined) => {
```
Comment (jimppa) [inline client/src/domain/event/EventHero.tsx]: Would it make sense to have this function as an utility function in some separate file?

File: client/src/domain/event/EventHero.tsx (hunk 7 above)
Comment (jorilindell) [reply client/src/domain/event/EventHero.tsx]: I forgot to change this. It used to have data that contains html. But not needed any more

**Pull request merged**